File: springboot/src/main/java/com/example/demo/service/ProductService.java
package com.example.demo.service;

import java.util.List;

import com.example.demo.binding.Product;

public interface ProductService {

	
	public String upsert(Product product);
	
	public Product getById(int pid);
	
	public List<Product> getAllProducts();
	
	public String deleteById(int pid);
}

File: ProductDemo/src/main/kotlin/com/example/service/ProductService.kt
package com.example.service





import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component
import org.springframework.stereotype.Service
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestBody
import java.util.*
import com.example.repository.ProductRepository
import com.example.model.Product


@Service
@Component
class ProductService(@Autowired private val productRepository : ProductRepository) {

    //get data from db


    fun getAllprod(): List<Product> {
        return productRepository.findAll();

    }

    fun getAllprodById(@PathVariable prodId: Int): Optional<Product> {
        return productRepository.findById(prodId);
    }

    fun CreateProd(@RequestBody prod: Product): Product {
        return productRepository.save(prod);
    }





}